spring-integration-redis/src/main/java/org/springframework/integration/redis/util/RedisLockRegistry.java

```diff
@@ -172,171 +158,65 @@ public RedisLockRegistry(RedisConnectionFactory connectionFactory, String regist
 	 * different {@link RedisLock} objects for same unlocked key.
 	 * @param useWeakReferences set to true for switch thread local weak references storage on, false by default
 	 * @since 4.0.7
+	 * @deprecated since 4.3.9
```

Comment:
`- the local optimization isn't used any more because of cluster-wide inconsistency`